src/test/java/org/commcare/formplayer/repo/FormSessionRepoTest.java

```diff
@@ -64,17 +64,19 @@ public void testSaveAndLoad() {
         formSessionRepo.saveAndFlush(session);
         entityManager.clear(); // clear the EM cache to force a re-fetch from DB
         SerializableFormSession loaded = JpaTestUtils.unwrapProxy(
-                formSessionRepo.getById(session.getId())
+                formSessionRepo.getReferenceById(session.getId())
         );
         assertThat(loaded).usingRecursiveComparison().ignoringFields("dateCreated",
                 "version").isEqualTo(session);
         Instant dateCreated = loaded.getDateCreated();
         assertThat(dateCreated).isNotNull();
-        assertThat(loaded.getVersion()).isEqualTo(1);
+        assertThat(loaded.getVersion()).isEqualTo(0);
 
+        // Make a edit to trigger version update
+        loaded.setInitLang("newlang");
         formSessionRepo.saveAndFlush(loaded);
         assertThat(loaded.getDateCreated()).isEqualTo(dateCreated);
-        assertThat(loaded.getVersion()).isEqualTo(2);
+        assertThat(loaded.getVersion()).isEqualTo(1);
```

Comment:
Can you explain these changes. They don't seem related to the Spring upgrade.

Reply:
I could not really find the related change documentation for this in hibernate but after upgrading to SB 3, the version doesn't increment unless you make a change to the object. My guess is that the change is caused by shifting to the Jakarta persistence apis from javax but I don't really have a supportive reference here that can explain this change.